src/cache_cf.cc

```diff
@@ -536,9 +546,17 @@ parseOneConfigFile(const char *file_name, unsigned int depth)
             /* Handle includes here */
             if (tmp_line_len >= 9 && strncmp(tmp_line, "include", 7) == 0 && xisspace(tmp_line[7])) {
                 err_count += parseManyConfigFiles(tmp_line + 8, depth + 1);
-            } else if (!parse_line(tmp_line)) {
-                debugs(3, DBG_CRITICAL, HERE << cfg_filename << ":" << config_lineno << " unrecognized: '" << tmp_line << "'");
-                ++err_count;
+            } else {
+                try {
+                    if (!parse_line(tmp_line)) {
+                        debugs(3, DBG_CRITICAL, ConfigPosition << "unrecognized: '" << tmp_line << "'");
+                        ++err_count;
+                    }
+                } catch (...) {
+                    // fatal for now
+                    debugs(3, DBG_CRITICAL, "configuration error: " << CurrentException);
+                    self_destruct();
```

Comment:
> Firstly it reverts a large amount of work done to make the -k parse feature maximize the amount of errors discovered.

I disagree: the old and the new code should behave similarly. The new try/catch near parse_line() results in catching and dying early within parseOneConfigFile(), while the old code would propagate the same exception up to parseConfigFile() and dying either. The 'trivial' errors reporting (which happen when parse_line() returns false because e.g., a parameter name is unknown), did not change: these errors were non-fatal and are still non-fatal. The new code should work just like the old code and report about all such cases. Basically, this PR only substitutes some self_destruct() with throw(), and in both cases '-k parse' aborts on the first error.

Reply:
This is the last point of disagreement here. We can resolve that with a file in test-suite/squidconf/ containing example config lines for all the directives affected by this change. It needs to test the expected and edge cases for the *old* configurable values so we can prove this change only adds new possibilities.

If in doing so you find some regressions from WARNING to ERROR those will need either fixing or documenting - decide on a case-by-case basis.

Reply:
> Amos: ... containing example config lines for all the directives affected by this change ...

If it is too difficult to collect _all_ of the directives that use modified time parsing routines, then adding at least one directive for each WARNING and non-fatal ERROR case (old and/or new) ought to be enough IMO. Please keep in mind that `test-suite/squidconf/` does not support test cases that are expected to fail so do not be tempted to add cases leading to Squid death.

If you add test cases that are expected to produce WARNINGs and ERRORs, please add squid.conf comments marking those test cases as such. Squid does not support proper testing of such cases (yet), but we should at least mark them when we add them.

Reply:
Done at 9455151.

> at least one directive for each WARNING and non-fatal ERROR case 

It is not possible to test related 'ERROR' cases, because all of them lead to Squid termination.
Both WARNING messages are displayed only when running '-k parse'.